axiom/nr_segment_children.h

```diff
@@ -165,36 +142,41 @@ static inline bool nr_segment_children_remove(nr_segment_children_t* children,
   }
 
   if (children->is_packed) {
-    size_t i;
+    size_t ix = child->child_ix;
     const size_t end = children->packed.count - 1;
+    nr_segment_t* temp;
 
-    if (child == children->packed.elements[end]) {
-      // The simple case: the child is the last element, so we can just
-      // decrement the count and we're done.
-      children->packed.count -= 1;
-      return true;
+    if (ix > end) {
+      return false;
     }
 
-    for (i = 0; i < end; i++) {
-      if (child == children->packed.elements[i]) {
-        nr_memmove(&children->packed.elements[i],
-                   &children->packed.elements[i + 1],
-                   sizeof(nr_segment_t*) * (children->packed.count - i - 1));
-        children->packed.count -= 1;
-        return true;
-      }
-    }
+    // Swap'n'Pop
+    temp = children->packed.elements[end];
+    temp->child_ix = ix;
+    children->packed.elements[ix] = temp;
+    children->packed.count -= 1;
+
   } else {
-    size_t index;
+    size_t index = child->child_ix;
+    nr_segment_t* temp;
```

Comment:
There's an opportunity here for a slight simplification of code since both the `if` and `else` statements require the `child->child_ix` value as well as the `nr_segment_t* temp` variable, so you could extract them out of the conditional statements and avoid code duplication. This would have the knock-on effect of also unifying the naming convention between the conditionals (`ix` vs `index`)

Reply:
Unfortunately, "ISO C90 forbids mixed declarations and code", so we cannot extract the definition of index to outside this structure, only the declaration. (Because the definition must happen after the `null` check).

I can move `size_t index` and` nr_segment_t* temp` to the top of the function if that is desired, or just unify the naming of "ix" vs "index".

Reply:
Probably not worth the trouble.